faust_avro/serializers.py

```
@@ -39,10 +39,22 @@ class AvroCodec:
     def __post_init__(self, schema=None):
         self.schema = fastavro.parse_schema(self.record.to_avro(self.registry))
 
+    async def compatible(self, registry: CSRC):
+        schema = json.dumps(self.record.to_avro(self.registry))
+        tasks = [registry.compatible(subject, schema) for subject in self.subjects]
+        compatible = await asyncio.gather(*tasks)
+        failed = [subj for subj, valid in zip(self.subjects, compatible) if not valid]
+        if failed:
+            print(f"{self.record.__module__}.{self.record.__name__} incompatible for: {failed}")
+
+        return all(compatible)
+
     async def register(self, registry: CSRC):
         schema = json.dumps(self.record.to_avro(self.registry))
         tasks = [registry.register(subject, schema) for subject in self.subjects]
-        self.schema_id, *_ = funcy.distinct(await asyncio.gather(*tasks))
+        self.schema_id, *ids = funcy.distinct(await asyncio.gather(*tasks))
+        assert not ids, "Schemas must share the same id across all subjects!"
```

Wouldn't it be better to raise an exception rather than a generic assert? That allows consumers of the library to handle the exception in the manner they choose.

---

The assert should never get tripped. Per https://docs.confluent.io/1.0/schema-registry/docs/api.html#post--subjects-(string-%20subject)-versions

> If the same schema is registered under a different subject, the same identifier will be returned. However, the version of the schema may be different under different subjects.

This was more defensive against someone co-opting the schema registry and altering it slightly in a cloud saas offering.